connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ConnectorConfigTest.java

```diff
@@ -147,10 +151,10 @@ public void singleTransform() {
         props.put("transforms.a.type", SimpleTransformation.class.getName());
         props.put("transforms.a.magic.number", "42");
         final ConnectorConfig config = new ConnectorConfig(MOCK_PLUGINS, props);
-        final List<Transformation<R>> transformations = config.transformations();
+        final List<TransformationStage<SinkRecord>> transformations = config.transformations();
         assertEquals(1, transformations.size());
-        final SimpleTransformation<R> xform = (SimpleTransformation<R>) transformations.get(0);
-        assertEquals(42, xform.magicNumber);
+        final TransformationStage<SinkRecord> xform = transformations.get(0);
+        assertEquals(42, xform.apply(DUMMY_RECORD).kafkaPartition().intValue());
```

Comment:
Both this refactor and a later refactor that I have planned are complicated by this sort of testing that depends on the _implementation_ of these classes (downcasting and field access). The new style depends on the _behavior_ of the configured instances, and works if the implementation details of the intermediate classes changes.

While it would be possible to preserve the original style by accessing the package-local PredicatedTransform delegate field, that is not possible for the next refactor I have planned. Rather than revisiting this test later, I figured I would fix it once and avoid the potential merge conflicts.

Reply:
Sounds good, I didn't really have any concerns with these changes; just curious about why they were being made.